symengine/tests/basic/test_functions.cpp

```diff
@@ -328,6 +330,14 @@ TEST_CASE("Sin: functions", "[functions]")
     r1 = rewrite_as_exp(sin(x));
     r2 = div(sub(exp(mul(I, x)), exp(mul(neg(I), x))), mul(integer(2), I));
     REQUIRE(eq(*r1, *r2));
+
+    r1 = rewrite_as_sin(sin(x));
+    r2 = sin(x);
+    REQUIRE(eq(*r1, *r2));
+
+    r1 = rewrite_as_cos(sin(x));
+    r2 = cos(unevaluated_expr(add(x, div(pi, integer(2)))));
+    REQUIRE(eq(*r1, *r2));
```

Comment:
it would also be nice to check that you can recover the original expression in each case by evaluating it, e.g. something like
```
    REQUIRE(eq(*parse(r1->__str__()), *sin(x)));
```
or maybe there is a better way to evaluate an UnevaluatedExpr than printing and re-parsing it?

Reply:
> maybe there is a better way to evaluate an UnevaluatedExpr than printing and re-parsing it?

There isn't any yet. SymPy has a `.doit()` method for this.

Reply:
I'll do as @lkeegan suggests, but I think we should have a function like `doit()` in the future. This would be useful also in other situations.